lib/semantics/check-do.cc

```diff
@@ -319,14 +319,42 @@ class DoConcurrentLabelEnforce {
   parser::CharBlock doConcurrentSourcePosition_{nullptr};
 };  // class DoConcurrentLabelEnforce
 
-using CS = std::vector<const Symbol *>;
+// Class for enforcing C1130
+class DoConcurrentVariableEnforce {
+public:
+  DoConcurrentVariableEnforce(
+      SemanticsContext &context, parser::CharBlock doConcurrentSourcePosition)
+    : context_{context}, doConcurrentSourcePosition_{
+                             doConcurrentSourcePosition} {}
 
-struct GatherSymbols {
-  CS symbols;
-  template<typename T> constexpr bool Pre(const T &) { return true; }
-  template<typename T> constexpr void Post(const T &) {}
-  void Post(const parser::Name &name) { symbols.push_back(name.symbol); }
-};
+  template<typename T> bool Pre(const T &) { return true; }
+  template<typename T> void Post(const T &) {}
```

Comment:
It seems like overkill creating a whole new visitor just for this check. It adds 700 new Walk functions to the generated code.

Did you consider adding a Checker (or adding to an existing one)? You would need Enter/Leave functions to manage a stack of active DO CONCURRENTs with DEFAULT(NONE) and Enter on parser::Name to perform the check.

Reply:
I'll look into adding another Checker.  There are additional semantic checks that we need to do for DO loops, which makes another checker more desirable.

Reply:
I created issue #624 for this.